topdown/topdown_test.go

```diff
@@ -3185,6 +3185,16 @@ func assertTopDownWithPath(t *testing.T, compiler *ast.Compiler, store storage.S
 
 	testutil.Subtest(t, note, func(t *testing.T) {
 		switch e := expected.(type) {
+		case Error:
```

Comment:
Nit: I'd rather remove this and use the expected error case below for checking errors. It would be a different story if this case was checking for value equality on the topdown.Error...but it's not.

Reply:
Errors from topdown contain the code and message: https://github.com/open-policy-agent/opa/blob/master/topdown/errors.go#L59. Also since these are surfaced as internal errors, it's less relevant to callers since it's not particularly actionable. This basically indicates that some unexpected error occurred. It's unlikely that callers can recover.

Reply:
I saw that while fixing the issue but they are concatenated so there is no straightforward way to properly compare code and message with the expected ones. The expected error is of the form:

	expectedResult := Error{Code: InternalErr, Message: "tls: private key does not match public key", Location: nil}

If you feel strongly about this I can change the tests not to check for the errors.

Reply:
The code under test doesn't set the code though, it's handled by the framework. I'd just assert on the message here.